### PR TITLE
Move the os-release file to /usr/lib and softlink

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit SvenDowideit@home.org.au

see #1241

### CODE REVIEW

File: cmd/console/console.go [hunk 1]
```diff
@@ -210,9 +210,10 @@ func modifySshdConfig() error {
 	return ioutil.WriteFile("/etc/ssh/sshd_config", []byte(sshdConfigString), 0644)
 }
 
+// https://www.freedesktop.org/software/systemd/man/os-release.html
```
Comment (ibuildthecloud) [inline cmd/console/console.go]: oh systemd and their silly standards...

File: cmd/console/console.go [hunk 2]
```diff
@@ -210,9 +210,10 @@ func modifySshdConfig() error {
...
 func writeOsRelease() error {
 	idLike := "busybox"
-	if osRelease, err := ioutil.ReadFile("/etc/os-release"); err == nil {
+	if osRelease, err := ioutil.ReadFile("/usr/lib/os-release"); err == nil {
```
Comment (ibuildthecloud) [inline cmd/console/console.go]: I feel bad changing `/usr/lib` as that is supposed to be read only.  According to the fake spec `/etc/os-release` should be read by application first.  So I'm guessing Docker is doing the wrong thing here if they complain `/usr/lib/os-release` isn't there if `/etc/os-release` is.

Comment (SvenDowideit) [inline cmd/console/console.go]: I picked this direction due to 

```
/etc/os-release should be a relative symlink to /usr/lib/os-release, to provide compatibility with applications only looking at /etc.
```

So - I guess its time for some larger context - why are we writing this file dynamically using `ros`?

Comment (joshwget) [inline cmd/console/console.go]: Doing it dynamically in `ros` simplifies console building. We should try to remove as many things as possible from the console building process so people can build their own more easily.

**Comment by SvenDowideit** [comment]
@joshwget @ibuildthecloud I'm leaning towards redoing this PR by moving the `os-release` file into the host FS, and then bind mounting it into the console containers. 

but that also makes me think we should do that the way the docker binaries are imported - bind mount into `/host`, and then softlink into the real place - that should allow us to sidestep the issue of `apt-get upgrade`.

**Comment by ibuildthecloud** [comment]
@SvenDowideit I agree bind mounting in is probably the best way to go.  Just to be clear on the host I'd still like to keep the file somewhere in `/usr` and then bind mount to something not `/usr/lib/os-release`.  This way we still keep all of ros really contained to `/usr` on the host.

File: scripts/package-rootfs [hunk 3]
```diff
@@ -22,6 +22,25 @@ if [ "$(docker info | grep 'Storage Driver: ' | sed 's/Storage Driver: //')" !=
     exit 0
 fi
 
+mkdir -p ${INITRD_DIR}/usr/rancher/
+cat <<HERE > ${INITRD_DIR}/usr/rancher/os-release
+NAME="RancherOS"
+VERSION=${VERSION}
+ID=rancheros
+ID_LIKE=rancher
```
Comment (joshwget) [inline scripts/package-rootfs]: Why is this "rancher"?

File: os-config.tpl.yml [hunk 4]
```diff
@@ -146,6 +146,7 @@ rancher:
       - /usr/bin/ros:/usr/bin/cloud-init-save:ro
       - /usr/bin/ros:/usr/sbin/netconf:ro
       - /usr/bin/ros:/usr/sbin/wait-for-docker:ro
+      - /usr/rancher:/usr/share/rancher:ro
```
Comment (joshwget) [inline os-config.tpl.yml]: I think this might make more sense under `system-volumes`. This will also cause upgrade issues for persistent consoles...

Comment (ibuildthecloud) [inline os-config.tpl.yml]: Wouldn't is be better to put `/usr/rancher` on the host at `/usr/share/rancher`?  It there a specific reason for mapping the folder to a different directory?

Comment (joshwget) [inline os-config.tpl.yml]: `/usr/rancheros` also feels like it would be better here.

File: scripts/package-rootfs (hunk 3 above)
Comment (SvenDowideit) [inline scripts/package-rootfs]: cos the host is not like anything :)

but y, its a random string

File: os-config.tpl.yml (hunk 4 above)
Comment (SvenDowideit) [inline os-config.tpl.yml]: I put them in different directories because when they were the same, they sometimes over-wrote each other and ended up empty :(

Comment (SvenDowideit) [inline os-config.tpl.yml]: @joshwget we use `rancher` everywhere else, so I'm going for consistent - we can revisit that in a few weeks when/if we have a rancher agent/server service

File: scripts/run [hunk 5]
```diff
@@ -8,6 +8,11 @@ source ./scripts/run-common
 
 while [ "$#" -gt 0 ]; do
     case $1 in
+        --iso)
```
Comment (SvenDowideit) [inline scripts/run]: and I also need a :paperclip:

**Pull request merged**